tensorflow/core/kernels/qr_op_float.cc

```diff
@@ -33,4 +33,20 @@ REGISTER_KERNEL_BUILDER(Name("Qr")
                         QrOp<float>);
 #endif
 
+#ifdef TENSORFLOW_USE_DIRECTML
+// For now, register QR as a CPU kernel for DML to follow the same registration
+// as the GPU. If the cuSolver bug ever get fixed we may want to implement our
+// own DML kernel to stay on par with CUDA, but in the meantime it's probably
+// not worth it to spend time on this kernel. Registering this kernel with CPU
```

Comment:
I'm not sure I follow why we're registering this for CPU. It's a slippery slope to register ops we are unlikely to implement to influence placement.

Generator ops (i.e. ops with no inputs that only produce outputs), like RandomStandardNormal, are placed on the same device as their consumers (assuming all consumers are on the same device). Are you trying to force the random ops to run on DML in some model?